composer/trainer/trainer.py

```diff
@@ -202,29 +206,58 @@ def __init__(
             else:
                 self.ddp_sync_strategy = DDPSyncStrategy(ddp_sync_strategy)
 
-        # TODO(#123): DeepSpeed still needs a precision context, but it's not completely clear how to
-        # handle this with our version of Pytorch
-        precision_context = self.device.precision_context if not self.deepspeed_enabled else cast(
-            Callable[..., ContextManager], contextlib.nullcontext)
+        if evaluators is not None:
+            for evaluator in evaluators:
+                if isinstance(evaluator.dataset, DataSpec):
+                    dataloader_spec = evaluator.dataset
+                else:
+                    dataloader_spec = DataSpec(evaluator.dataset)
+                evaluator.dataset = DDPDataLoader(dataloader_spec.dataloader)
+        self.evaluators = evaluators
+
+        if eval_dataloader is not None:
+            if isinstance(eval_dataloader, DataSpec):
+                eval_dataloader_spec = eval_dataloader
+            else:
+                eval_dataloader_spec = DataSpec(eval_dataloader)
+            default_evaluator = Evaluator(label="eval_dataset",
+                                          dataset=DDPDataLoader(eval_dataloader_spec.dataloader),
+                                          metrics=None,
+                                          validate_every_n_batches=validate_every_n_batches,
+                                          validate_every_n_epochs=validate_every_n_epochs)
+            if self.evaluators is not None:
+                self.evaluators.append(default_evaluator)
+            else:
+                self.evaluators = [default_evaluator]
 
         if not isinstance(train_dataloader, DataSpec):
             train_dataloader = DataSpec(train_dataloader)
         train_dataloader.dataloader = DDPDataLoader(train_dataloader.dataloader)
-        if not isinstance(eval_dataloader, DataSpec):
-            eval_dataloader = DataSpec(eval_dataloader)
-        eval_dataloader.dataloader = DDPDataLoader(eval_dataloader.dataloader)
 
-        self.state = State(
-            max_epochs=max_epochs,
-            algorithms=algorithms,
-            callbacks=callbacks,
-            model=model,
-            grad_accum=grad_accum,
-            precision=precision,
-            precision_context=precision_context,
-            train_dataloader=train_dataloader,
-            eval_dataloader=eval_dataloader,
-        )
+        # do a check here to make sure there is at least one validation set
+        if self.evaluators is None or len(self.evaluators) == 0:
+            raise ValueError('At least one validation set should be used and passed in through ',
```

Comment:
I'd like to allow no evaluators (e.g. in tests), so an error is a bit harsh, but I think a warning would be fine here.